### PR TITLE
Fix compiler warnings

### PR DESCRIPTION
**Pull request by adolfo-rt**
I attempted to build moveit_ros from source with rather strict build flags (many -Werror settings + cppckeck static code analyzer). This PR gathers the more relevant issues that surfaced. I opted for multiple commits to identify fixes by type and facilitate review. If the review is successful, I could squash them together if you prefer to keep the repo history succinct.

### CODE REVIEW

File: move_group/src/move_group_capability.cpp [hunk 1]
```diff
@@ -77,7 +77,7 @@ void move_group::MoveGroupCapability::convertToMsg(const robot_trajectory::Robot
 void move_group::MoveGroupCapability::convertToMsg(const std::vector<plan_execution::ExecutableTrajectory> &trajectory,
                                                    moveit_msgs::RobotState &first_state_msg, moveit_msgs::RobotTrajectory &trajectory_msg) const
 {
-  if (trajectory.size() > 1)
+  if (!trajectory.empty())
```
Comment (acornacorn) [inline move_group/src/move_group_capability.cpp]: This is a logic change.  Was this intentional?  I think this change may be incorrect.

Comment (adolfo-rt) [inline move_group/src/move_group_capability.cpp]: You are right. I'll undo the commit.

**Comment by adolfo-rt** [comment]
The commit entitled  "Address [cppcheck: stlSize] warning." was incorrect, and has been removed from the PR. Thanks @acornacorn for spotting it.

**Comment by acornacorn** [comment]
+1.  Nice work!

**Comment by isucan** [comment]
+1 Thank you!

**Comment by davetcoleman** [comment]
+1

**Pull request merged**